extension/chrome/elements/compose-modules/formatters/encrypted-mail-msg-formatter.ts

```diff
@@ -22,19 +22,77 @@ import { AcctStore } from '../../../../js/common/platform/store/acct-store.js';
+      // pwdRecipients that have their personal link
+      const pwdRecipients = Object.keys(uploadedMessageData.emailToExternalIdAndUrl ?? {}).filter(email => !pubkeys.some(p => p.email === email));
       newMsg.pwd = undefined;
-      return await this.sendablePwdMsg(newMsg, pubkeys, { msgUrl, externalId }, signingPrv); // encrypted for pubkeys only, pwd ignored
-    } else if (this.richtext) { // rich text: PGP/MIME - https://tools.ietf.org/html/rfc3156#section-4
+      const collectedAttachments = await this.view.attachmentsModule.attachment.collectEncryptAttachments(pubkeys);
+      const pubkeyRecipients: { [type in RecipientType]?: EmailParts[] } = {};
+      for (const [key, value] of Object.entries(newMsg.recipients)) {
+        if (['to', 'cc', 'bcc'].includes(key)) {
+          const sendingType = key as RecipientType;
+          pubkeyRecipients[sendingType] = value?.filter(emailPart => pubkeys.some(p => p.email === emailPart.email)
+            // pwd recipients that have no personal links go to legacy message
+            || (uploadedMessageData.emailToExternalIdAndUrl || {})[emailPart.email] === undefined);
+        }
+      }
+      const uniquePubkeyRecipientToAndCCs = Value.arr.unique((pubkeyRecipients.to || []).concat(pubkeyRecipients.cc || [])
+        .map(recipient => recipient.email.toLowerCase()));
+      // pubkey recipients should be able to reply to "to" and "cc" pwd recipients
+      const replyTo = (newMsg.recipients.to ?? []).concat(newMsg.recipients.cc ?? [])
+        .filter(recipient => !uniquePubkeyRecipientToAndCCs.includes(recipient.email.toLowerCase()));
+      const msgs: SendableMsg[] = [];
+      if (pubkeyRecipients.to?.length || pubkeyRecipients.cc?.length || pubkeyRecipients.bcc?.length) {
+        const pubkeyMsgData = {
+          ...newMsg,
+          recipients: pubkeyRecipients,
+          // brackets are required for test emails like '@test:8001'
+          replyTo: replyTo.length ? `${Str.formatEmailList([newMsg.from, ...replyTo], true)}` : undefined
+        };
+        msgs.push(await this.sendablePwdMsg(
+          pubkeyMsgData,
+          pubkeys,
+          { msgUrl: uploadedMessageData.url, externalId: uploadedMessageData.externalId },
+          collectedAttachments,
+          signingKey?.key)
+        );
+      }
+      // adding individual messages for each recipient that doesn't have a pubkey
+      for (const recipientEmail of pwdRecipients) {
+        const { url, externalId } = uploadedMessageData.emailToExternalIdAndUrl![recipientEmail];
+        const foundParsedRecipient = (newMsg.recipients.to ?? []).concat(newMsg.recipients.cc ?? []).concat(newMsg.recipients.bcc ?? []).
+          find(r => r.email.toLowerCase() === recipientEmail.toLowerCase());
+        // todo: since a message is allowed to have only `cc` or `bcc` without `to`, should we preserve the original placement(s) of the recipient?
```

Comment:
should I preserve the placement of the individual recipient (to, cc or bcc) when sending an individual message. It works fine with Gmail -- a message can have only `bcc` filled, for example.

Reply:
ping @tomholub

Reply:
Unsure. I'll let you decide, you can do whatever is simpler.